v1-2/learn/how-to-use-java-interoperability.md

```diff
@@ -0,0 +1,939 @@
+
+The `java.util.zip.ZipFile` class is used to read entries in a zip file. There are many constructors in this class. Here, the constructor that takes the file name as an argument is used. 
+
+```java
+public ZipFile(String name) throws IOException
+```
+
+Since this Java constructor throws a checked exception,  the `newZipfile` Ballerina function returns `ZipFile` instances or an error. 
+
+```ballerina
+import ballerinax/java;
+
+function newZipFile(handle filename) returns handle | error = @java:Constructor {
+   class: "java.util.zip.ZipFile",
+   paramTypes: ["java.lang.String"]
+} external;
+
+public function main() {
+   handle|error zipFile = newZipFile(java:fromString("some_file.zip"));
+}
+```
+
+#### Mapping a Java exception to a Ballerina error value
+Now, let’s briefly look at how a Java exception is converted to a Ballerina error value at runtime. A Ballerina error value contains three components: a reason, a detail, and stack trace. 
+
+The `reason`:
+	* This is a string identifier for the category of error.
+	* In this case, reason value is set to the fully qualified Java class name of the exception. 
+		* Unchecked: Class name of of the thrown unchecked exception
+		* Checked: Class name of the exception that is declared in the method signature
+The `detail`:
+	* The `message` field is set to `e.getMessage()`.
+	* The `cause` field is set to the Ballerina error that represents this Java exception’s cause.
+
+### Null safety
+Ballerina provides strict null safety compared to Java with optional types.  The Java null reference can be assigned to any reference type. However, in Ballerina, you cannot assign the nil value to a variable unless the variable’s type is an optional type. 
+
+As explained above, Ballerina handle values cannot be created in Ballerina code. They are created and returned by foriegn functions and a variable of the handle type refers to a Java reference value. Since Java null is also a valid reference value, this variable can refer to a Java null value.
+
+Let’s look at an example that deals with Java null. The following code uses the `peek` method in the `ArrayDeque` class. Peek retrieves but does not remove the head of the queue or returns null if the queue is empty. 
+
+```ballerina
+import ballerinax/java;
+
+function newArrayDeque() returns handle = @java:Constructor {
+   class: "java.util.ArrayDeque"
+} external;
+
+function peek(handle receiver) returns handle = @java:Method {
+   class: "java.util.ArrayDeque"
+} external;
+
+// Linked with the java.lang.Object.toString() method in Java
+function toString(handle objInstance) returns handle = @java:Method {
+   class: "java.lang.Object"
+} external;
+
+public function main() {
+   handle arrayDeque = newArrayDeque();
+   handle element = peek(arrayDeque);
+   Handle str = toString(element);
+}
+```
+
+Since the queue is empty in this case, `peek` should return null. I.e., `element` should refer to Java null.  The output of this program will be as follows.
```

Comment:
```suggestion
Since the queue is empty in this case, `peek` should return null i.e., `element` should refer to Java null.  The output of this program will be as follows.
```